### PR TITLE
Merge VAOS and Mobile Appointment Authorization Policy

### PR DESCRIPTION
**Pull request by Tonksthebear**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
This serves to further merge shared logic between web and mobile.

From our understanding, web was performing the facility check on the front end, so this moved it to the backend.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9617

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by jperk51** [comment]
> Could someone from web confirm that moving the facilities check to the backend won't cause issues?

I shouldn't make a difference. They check the user info on the FE so it won't even make a call to the appts endpoint if the user has no facilities.

**Review by jperk51 — APPROVED** [review]

**Comment by simiadebowale** [comment]
@jperk51 - correct, this shouldn't break anything on web

**Comment by stevenjcumming** [comment]
@Tonksthebear there are failing tests

**Comment by simiadebowale** [comment]
@Tonksthebear - I'm not sure if all consumers of our service do a facilities check before calling our endpoint. To be safe, let's assume they don't

**Comment by Tonksthebear** [comment]
@simiadebowale I can do that. Do you know which ones for sure do check for facilities?

**Comment by stevenjcumming** [comment]
@Tonksthebear can you have a teammate review/approve, then I'll review/approve

**Review by jperk51 — COMMENTED** [review]
This looks good but also make sure to update the mobile authorized services check for [appointments](https://github.com/department-of-veterans-affairs/vets-api/blob/f568ea641b63f5c6d7330e4397569b615a48f3e7/modules/mobile/app/models/mobile/v0/user_accessible_services.rb#L24C11-L24C71).

**Comment by Tonksthebear** [comment]
@jperk51 Updated the policy

@simiadebowale I got confirmation from Corey but would like your confirmation as well that removing the Flipper code is ok. Our thinking was scheduling should not be a requirement to view appointments

File: modules/vaos/app/policies/vaos_policy.rb
Comment (aherzberg) [inline modules/vaos/app/policies/vaos_policy.rb:5]: this seems right to remove but have we confirmed with VAOS that its won't cause any issues?

**Review by aherzberg — APPROVED** [review]
LGTM

**Review by jperk51 — APPROVED** [review]
lgtm, but I would like to have @aherzberg and @cferris32 to approve this also

File: modules/mobile/app/models/mobile/v0/user_accessible_services.rb
Comment (aherzberg) [inline modules/mobile/app/models/mobile/v0/user_accessible_services.rb:24]: I think this will break the intended logic. The intention for that facilities access method was to return a specific error message that FE can key off of and show a banner about how they don't have facilities. If this returns false, then FE won't get a chance to get that error message.

**Review by aherzberg — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (aherzberg) [reply .github/CODEOWNERS:28]: the reason I was thinking we need ownership as well is FE will be keying off of a specific error message for that no facilities error so if it gets changed in this file and we aren't aware, that'd prevent the banner from ever showing.

Comment (Tonksthebear) [reply .github/CODEOWNERS:28]: Should I remove us @jperk51 or is it worth keeping us on for the reason Andrew specified?

File: modules/vaos/app/policies/vaos_policy.rb
Comment (Tonksthebear) [reply modules/vaos/app/policies/vaos_policy.rb:5]: Waiting for confirmation

File: modules/mobile/app/models/mobile/v0/user_accessible_services.rb
Comment (jperk51) [reply modules/mobile/app/models/mobile/v0/user_accessible_services.rb:24]: Was this choice discussed somewhere? The original intention behind the vets-api policy update was so the mobile authed services checks act more accurately and so that we don't make calls to apis we know will fail.

Comment (Tonksthebear) [reply modules/mobile/app/models/mobile/v0/user_accessible_services.rb:24]: So we want to allow access even though it's doomed to fail? Is that better than just not letting them have access at all?

Comment (jperk51) [reply modules/mobile/app/models/mobile/v0/user_accessible_services.rb:24]: The driving factor behind adding this check was when the FE added appts overview to the home screen it massively spiked the number of calls being made to VAOS and adding the facilities check, which web has on the FE, to the vets-api policy (and our authed services by proxy) would lower the number of calls being made. If there was a discussion I'm not aware of where mobile FE/UX decided they wanted to specific isolate this exact reason for failure then we can let them through but it does go against our pattern a bit and also goes against the purpose of the policy updating epic.

Comment (jperk51) [reply modules/mobile/app/models/mobile/v0/user_accessible_services.rb:24]: Okay cool. Let me take a look at those threads real quick. It's likely not an issue to let them through. You're correct that it stops upstream calls still, but generally the purpose of the authed services endpoint is to stop calls from being made to vets-api unnecessarily

Comment (jperk51) [reply modules/mobile/app/models/mobile/v0/user_accessible_services.rb:24]: So looking at those threads it looks like no actual decision was made for if we want a banner or not or what the banner should say. @Tonksthebear go ahead and remove this for now, sorry for the extra headache. I'll do some circling on the banner and depending on what comes out of it we might end up adding this back.

File: .github/CODEOWNERS
Comment (jperk51) [reply .github/CODEOWNERS:28]: You can add us for now.

**Review by jperk51 — DISMISSED** [review]

**Review by aherzberg — DISMISSED** [review]
LGTM

**Comment by Tonksthebear** [comment]
@stevenjcumming approved

**Review by stevenjcumming — APPROVED** [review]
There's an error in the codeowners file, but I'm going to open a PR to fix that. So you can ignore that

**Review by stevenjcumming — CHANGES_REQUESTED** [review]
Sorry I was wrong... `va-mobile-api-engineers` doesn't exist. Please use another team like `va-api-engineers`

**Comment by Tonksthebear** [comment]
@stevenjcumming Done!

**Review by stevenjcumming — APPROVED** [review]
Thanks for fixing the codeowners. You can always change it to a more appropriate team later.

File: .github/CODEOWNERS
Comment (jperk51) [reply .github/CODEOWNERS:28]: our team is just mobile-api-team

**Review by aherzberg — APPROVED** [review]
LGTM

**Pull request merged**
